packages/main/src/TimePicker.js

```diff
@@ -0,0 +1,579 @@
+import PopoverHorizontalAlign from "./types/PopoverHorizontalAlign.js";
+import ResponsivePopover from "./ResponsivePopover.js";
+import PopoverPlacementType from "./types/PopoverPlacementType.js";
+import TimePickerTemplate from "./generated/templates/TimePickerTemplate.lit.js";
+import {
+	TIMEPICKER_HOURS_LABEL,
+	TIMEPICKER_MINUTES_LABEL,
+	TIMEPICKER_SECONDS_LABEL,
+	TIMEPICKER_PERIODS_LABEL,
+} from "./generated/i18n/i18n-defaults.js";
+import {
+	isLeft,
+	isRight,
+	isTabNext,
+	isTabPrevious,
+} from "../../base/src/events/PseudoEvents.js";
+
+// Styles
+import TimePickerCss from "./generated/themes/TimePicker.css.js";
+import ResponsivePopoverCommonCss from "./generated/themes/ResponsivePopoverCommon.css.js";
+
+/**
+ * @public
+ */
+const metadata = {
+	tag: "ui5-timepicker",
+	properties: /** @lends sap.ui.webcomponents.main.TimePicker.prototype */ {
+		/**
+		 * Defines a formatted time value.
+		 *
+		 * @type {string}
+		 * @defaultvalue ""
+		 * @public
+		 */
+		value: {
+			type: String,
+			defaultValue: "",
+		},
+
+		/**
+		 * Determines the format, displayed in the input field.
+		 *
+		 * @type {string}
+		 * @defaultvalue ""
+		 * @public
+		 */
+		formatPattern: {
+			type: String,
+			defaultValue: "HH:mm:ss",
+		},
+
+		/**
+		 * Visualizes the validation state of the Web Component, for example
+		 * <code>Error</code>, <code>Warning</code> and
+		 * <code>Success</code>.
+		 *
+		 * @type {string}
+		 * @defaultvalue "None"
+		 * @public
+		 */
+		valueState: {
+			type: ValueState,
+			defaultValue: ValueState.None,
+		},
+
+		/**
+		 * Defines a short hint, intended to aid the user with data entry when the
+		 * <code>ui5-datepicker</code> has no value.
+		 *
+		 * <b>Note:</b> When no placeholder is set, the format pattern is displayed as a placeholder.
+		 * Passing an empty string as the value of this property will make the <code>ui5-time-picker</code> appear empty - without placeholder or format pattern.
+		 *
+		 * @type {string}
+		 * @defaultvalue undefined
+		 * @public
+		 */
+		placeholder: {
+			type: String,
+			defaultValue: undefined,
+		},
+
+		/**
+		 * Determines whether the <code>ui5-datepicker</code> is displayed as disabled.
+		 *
+		 * @type {boolean}
+		 * @defaultvalue false
+		 * @public
+		 */
+		disabled: {
+			type: Boolean,
+		},
+
+		/**
+		 * Determines whether the <code>ui5-datepicker</code> is displayed as readonly.
+		 *
+		 * @type {boolean}
+		 * @defaultvalue false
+		 * @public
+		 */
+		readonly: {
+			type: Boolean,
+		},
+
+		_isPickerOpen: {
+			type: Boolean,
+			noAttribute: true,
+		},
+
+		_respPopover: {
+			type: Object,
+		},
+
+		_hours: {
+			type: String,
+		},
+
+		_minutes: {
+			type: String,
+		},
+
+		_seconds: {
+			type: String,
+		},
+
+		isTwelveHoursFormat: {
+			type: Boolean,
+		},
+	},
+	slots: /** @lends sap.ui.webcomponents.main.TimePicker.prototype */ {
+		//
+	},
+	events: /** @lends sap.ui.webcomponents.main.TimePicker.prototype */ {
+		//
```

Comment:
Here we need an event called "change" to notify when we pick up a new time and to be complaint with all input based components from the other hand

Reply:
Done